pisa/utils/hypersurface.py

```diff
@@ -0,0 +1,1927 @@
+        This is the overall intercept, plus the coefficients for each individual param
+        '''
+        return int( 1 + np.sum([ param.num_fit_coeffts for param in list(self.params.values()) ]) )
+
+
+    @property
+    def fit_coeffts(self) :
+        '''
+        Return all coefficients, in all bins, as a single array
+        This is the overall intercept, plus the coefficients for each individual param
+        Dimensions are: [binning ..., fit coeffts]
+        '''
+        
+        array = [self.intercept]
+        for param in list(self.params.values()) :
+            for i in range(param.num_fit_coeffts) :
+                array.append( param.get_fit_coefft(coefft_idx=i) )
+        array = np.stack(array,axis=-1)
+        return array
+
+
+    @property
+    def fit_coefft_labels(self) :
+        '''
+        Return labels for each fit coefficient
+        '''
+        return ["intercept"] + [ "%s p%i"%(param.name,i) for param in list(self.params.values()) for i in range(param.num_fit_coeffts) ]
+
+
+    @property
+    def serializable_state(self):
+        """
+        OrderedDict containing savable state attributes
+        """
+
+        if self._serializable_state is None: #TODO always redo?
+
+            state = collections.OrderedDict()
+
+            state["_initialized"] = self._initialized
+            state["binning"] = self.binning.serializable_state
+            state["initial_intercept"] = self.initial_intercept
+            state["intercept"] = self.intercept
+            state["intercept_sigma"] = self.intercept_sigma
+            state["fit_complete"] = self.fit_complete
+            state["fit_info_stored"] = self.fit_info_stored
+            state["fit_maps_norm"] = self.fit_maps_norm
+            state["fit_maps_raw"] = self.fit_maps_raw
+            state["fit_chi2"] = self.fit_chi2
+            state["fit_cov_mat"] = self.fit_cov_mat
+            state["fit_method"] = self.fit_method
+            state["using_legacy_data"] = self.using_legacy_data
+
+            state["params"] = collections.OrderedDict()
+            for name,param in list(self.params.items()) :
+                state["params"][name] = param.serializable_state
+
+            self._serializable_state = state
+
+        return self._serializable_state 
+
+
+    @classmethod
+    def from_state(cls, state):
+        """
+        Instantiate a new object from the contents of a serialized state dict
+
+        Parameters
+        ----------
+        resource : dict
+            A dict
+
+        See Also
+        --------
+        to_json
+        """
+
+        #
+        # Get the state
+        #
+
+        # If it is not already a a state, alternativey try to load it in case a JSON file was passed
+        if not isinstance(state,collections.Mapping) :
+            try :
+                state = from_json(state)
+            except:
+                raise IOError("Could not load state")
```

Comment:
You are not helping here. You are masking the real exception (you use the catch-all) by doing this, so you are actually making it harder for the user to understand why the code fails here by doing this try / except. Just let it fail if it fails, the traceback will say this is where it failed, but also include the original error message. If you are desperate to include more information (which you do not currently do), then use `logging.error(...)` ... example of something that could help the user would be to print out the `state` that failed to load (but that could be very long output, and actually just be unhelpful anyway).

Reply:
Fixed. Now just let it fail with out try-except